src/utilities/TagRandomizer.py

```diff
@@ -0,0 +1,119 @@
+        
+        self.rngLog.info(f"Initializing the Tag RNG class with parameters: dict path: {self.dict_path} dict_size: {self.dict_size} max tags: {self.max_tags} min_tags{self.min_tags}.")
+        
+    def getRandomTags(self, exact=0) -> (str, int):
+        """Generates a string of comma-separated random tags extracted from the
+           provided source dictionary.  Selects a random number of tags between
+           the specified min and max values, defaulting to the init values.
+
+           Input: self - Pointer to the current object instance.
+                  exact - an exact number of tags to return (instead of min/max).
+              
+           Output: str - A comma-separated string of tags, doesn't assume the
+                         parent added its own leading comma.
+                   int - How many random tags were added to the prompt.
+        """
+        tag_count = 0
+        tag_list  = ""
+        
+        #Users trying to be cute will get the default result.
+        if exact > 0:
+            tag_count = exact
+        else:
+            tag_count = rand.randint(self.min_tags, self.max_tags)
+            
+        for word in range(0, tag_count):
+             
+             tag = lc.getline(self.dict_path, rand.randint(0, self.dict_size))
+             
+             if tag_list.find(tag) > -1:
+             
+                self.rngLog.debug(f"Found duplicate tag {tag} from tag_list {tag_list}, attempting to get another")
+                
+                for retry in range (0, self.rand_retries):
+                
+                    tag = lc.getline(self.dict_path, rand.randint(0, self.dict_size))
+                    
+                    if tag_list.find(tag) == -1:
+                    
+                        self.rngLog.debug(f"Found new unique tag")
+                        break
+                    
+                self.rngLog.debug(f"Exiting de-dupe loop with tag {tag}.")
+             #The getline function includes getting the separator.
+             tag_list += (', ' + tag.rstrip(os.linesep))
+        
+        #Note: Embedded fields (like how tags are displayed in the Discord post)
+        #only allow up to self.post_limit characters, meaning we must truncate
+        #the list to avoid exceptions in post-processing.
+        if len(tag_list) > self.post_limit:
+            #Find the last instance of a comma before the limit, and remove
+            #everything following.
+            index = tag_list.rfind(',', 0, self.post_limit)
+            
+            #If no comma is found at all, somehow, then we're forced to do a
+            #blunt truncate.
+            if index == -1:
+            
+                tag_list = tag_list[0:self.post_limit]
+                self.rngLog.warning(f"Tag list lacks comma separators and was bluntly truncated to {self.post_limit}.  This may still fail to post!")
+                
+            else:
```

Comment:
In my opinion it would be easier if you could just work with tag_list as an actual list instead of a string. You could use more helpful functions such as split which basically does this work for you.

Reply:
Yeah; the problem is that StableDiffusion only takes a string input for this parameter, so I'd have to convert types twice just to do a limit check (and back).

At least this way I do have a relatively easy way to guarantee the string length, instead of trying to sum the length of each element in a list (up to a set value).

Reply:
Seems lazy from whoever developed StableDiffusion to not accept lists. Oh well

Reply:
It's probably because it's passed over a POST call, but it would be nice if a list worked instead.